spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkBatchQueryScan.java

```diff
@@ -158,6 +163,26 @@ public void filter(Predicate[] predicates) {
     }
   }
 
+  protected Map<String, DeleteFileSet> dataToFileScopedDeletes() {
```

Comment:
[doubt] why do we need this whole hash-map of all the files with deletes being broadcasted from driver to executor ? since they are being anyways derived from scanTasks and spark executor ideally should have scanTask() so can we not create a local hashmap within executor and merge ? Am i missing something here ? 


[a bit orthogonal] can we put an estimate on the size of the HM ? if it goes very high it can fail the query ?  i think the size is 8GB if IIRC.